src/nodes.js

```diff
@@ -143,10 +154,44 @@ var registerChild = function(parent, key, child) {
 };
 
 
+/**
+ * Marks the key map for a given Element as invalid. It will be cleaned the
+ * next time it is needed.
+ * @param {!Element} el
+ */
+var invalidateKeyMap = function(el) {
+  getKeyMap(el).isValid = false;
+};
+
+
+/**
+ * Cleans a key map for a given Element, removing any nodes that are no longer
+ * in the DOM.
+ * @param {!Element} el
+ */
+var cleanKeyMap = function(el) {
+  var keyMap = getData(el).keyMap;
+
+  if (!keyMap || keyMap.isValid) {
+    return;
+  }
+
+  for (var key in keyMap) {
+    if (!keyMap[key].parentNode) {
+      delete keyMap[key];
```

Comment:
I don't think that it is a good idea. Holding on to a single node is okay, but you could end up holding on to very large subtrees. Additionally, those subtrees could hold on to large amounts of JavaScript due to event listeners.

For example, the application might want to key some views that it has for quick switching but may remove those views after some inactivity. If we end up holding on to even a handful of them, we may increase the memory usage of the application by an order of magnitude.

I think even this approach probably is not a great idea (though you only hold on to stuff for one pass).